connectivity/FEATURE_BLE/include/ble/gap/ChainableGapEventHandler.h

```diff
@@ -0,0 +1,274 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2006-2020 ARM Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_CHAINABLEGAPEVENTHANDLER_H_
+#define MBED_CHAINABLEGAPEVENTHANDLER_H_
+
+#include "ble/Gap.h"
+#include "ble/common/ChainableEventHandler.h"
+
+/**
+ * Gap::EventHandler implementation that allows the application
+ * to register multiple separate EventHandlers to be called when
+ * Gap events happen.
+ */
+class ChainableGapEventHandler : public ble::Gap::EventHandler,
+                                 public ChainableEventHandler<ble::Gap::EventHandler>
+{
+
+public:
+
+    ChainableGapEventHandler() { }
+
+    virtual ~ChainableGapEventHandler() { }
+
+    /**
+     * Called when an advertising device receive a scan response.
+     *
+     * @param event Scan request event.
+     *
+     * @version: 5+.
+     *
+     * @see AdvertisingParameters::setScanRequestNotification().
+     */
+    void onScanRequestReceived(const ble::ScanRequestEvent &event) override {
```

Comment:
For documentation of the event handler, I'd just reference the parent's documentation so we don't have duplicates that requires update. 

You can either remove the command as the doxygen flag `INHERIT_DOCS` is set to `YES` ([ref](https://www.doxygen.nl/manual/config.html#cfg_inherit_docs)) or  use [`@copydoc`](https://www.doxygen.nl/manual/commands.html#cmdcopydoc)